lib/Controller/MessageApiController.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+namespace OCA\Mail\Controller;
+
+use OCA\Mail\Db\LocalMessage;
+use OCA\Mail\Exception\ClientException;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Exception\UploadException;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Service\AliasesService;
+use OCA\Mail\Service\Attachment\AttachmentService;
+use OCA\Mail\Service\Attachment\UploadedFile;
+use OCA\Mail\Service\OutboxService;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\NoAdminRequired;
+use OCP\AppFramework\Http\Attribute\NoCSRFRequired;
+use OCP\AppFramework\Http\Attribute\UserRateLimit;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\DB\Exception;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class MessageApiController extends OCSController {
+
+	private ?string $userId;
+
+	public function __construct(
+		string $appName,
+		$UserId,
+		IRequest $request,
+		private AccountService $accountService,
+		private AliasesService $aliasesService,
+		private AttachmentService $attachmentService,
+		private OutboxService $outboxService,
+		private LoggerInterface $logger,
+		private ITimeFactory $time,
+	) {
+		parent::__construct($appName, $request);
+		$this->userId = $UserId;
+	}
+
+	/**
+	 * @param int $accountId
+	 * @param string $fromEmail
+	 * @param string $subject
+	 * @param string $body
+	 * @param bool $isHtml
+	 * @param array $to
+	 * @param array $cc
+	 * @param array $bcc
+	 * @param array $references
+	 * @return DataResponse
+	 */
+	#[UserRateLimit(limit: 5, period: 100)]
+	#[NoAdminRequired]
+	#[NoCSRFRequired]
+	public function send(
+		int $accountId,
+		string $fromEmail,
+		string $subject,
+		string $body,
+		bool $isHtml,
+		array $to,
+		array $cc = [],
+		array $bcc = [],
+		array $references = [],
+	): DataResponse {
```

Comment:
Instead of using individual parameters for all possible options I would recommend changing this to a message Json object.

Something like... 

Function...
```
public function send(int $accountId, string $message, array $options) {

}

```

Message Object...
```
{
    "from": [
        {
            "label": "User One via Nextcloud",
            "email": "ncdev@ksamail.ca"
        }
    ],
    "to": [
        {
            "label": "User Two",
            "email": "user2@ksamail.ca"
        }
    ],
    "replyTo": [
        {
            "label": "User One",
            "email": "user1@ksamail.ca"
        }
    ],
    "cc": [],
    "bcc": [],
    "subject": "Invitation: Recurrance Test",
    "dateInt": 1718673419,
    "flags": {
        "hasAttachments": false,
        "important": false
    },
    "hasHtmlBody": true,
    "body": "==== Email Body =====",
    "attachments": [
        {
            "id": "2",
            "fileName": "event.ics",
            "mime": "text\/calendar",
            "size": 1566,
            "cid": null,
        }
    ],
}
```

Reply:
We would lose all typing